lib/Sema/CodeSynthesis.cpp

```diff
@@ -1081,9 +1083,11 @@ static void synthesizeObservedSetterBody(AccessorDecl *Set,
   // TODO: check the body of didSet to only do this load (which may call the
   // superclass getter) if didSet takes an argument.
   VarDecl *OldValue = nullptr;
-  if (VD->getDidSetFunc()) {
-    Expr *OldValueExpr
-      = createPropertyLoadOrCallSuperclassGetter(Set, VD, target, Ctx);
+  AccessorDecl *didSetFunc =
+      storageToUse ? Set->getStorage()->getDidSetFunc() : VD->getDidSetFunc();
+  if (didSetFunc) {
+    Expr *OldValueExpr = createPropertyLoadOrCallSuperclassGetter(
+        Set, Set->getStorage(), target, Ctx);
```

Comment:
We don't want to use the lazy var's backing storage here because it has an optional type, so you will get an error: `value of optional type 'T?' must be unwrapped to a value of type 'T'`